### PR TITLE
Add a wrapping component to translate props

### PR DESCRIPTION
**Pull request by uniphil**
We need something like this to translate props, like the `alt` prop on ``. See the tests for example usage.

### CODE REVIEW

**Comment by uniphil** [comment]
trying another technique

File: app/scripts/components/misc/t-set-child-props.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import warn from '../../utils/warn';
+import { translate } from './t';
+
+const TSetChildProps = React.createClass({
+  propTypes: {
+    children: React.PropTypes.node.isRequired,
+  },
+
+  getInitialState() {
+    return {lang: 'en'};
```
Comment (esoliani) [inline app/scripts/components/misc/t-set-child-props.jsx:14]: this should be done using the language store. shouldn't it?

File: app/scripts/components/misc/t-set-child-props.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  },
+
+  translateObj(obj) {
+    return translate(this.state.lang, obj.k, obj.i);
+  },
+
+  getTranslatedProps(child) {
+    return Object.keys(child.props)
+     .filter(propName => this.isTranslateObj(child.props[propName]))
```
Comment (esoliani) [inline app/scripts/components/misc/t-set-child-props.jsx:54]: how do you do this `props[propName]` without lint complains about?

File: app/scripts/components/misc/t-set-child-props.jsx (hunk 1 above)
Comment (uniphil) [inline app/scripts/components/misc/t-set-child-props.jsx:14]: yep, however the language store doesn't exist yet (#30 adds it)

File: app/scripts/components/misc/t-set-child-props.jsx (hunk 2 above)
Comment (uniphil) [inline app/scripts/components/misc/t-set-child-props.jsx:54]: The lint allows it when `propName` is a variable, not a string

**Pull request merged**